src/ImageLoaders/MzExe/PharLapExtender.cs

```diff
@@ -46,15 +51,22 @@ public override Address PreferredBaseAddress
 
         public override Program Load(Address addrLoad)
         {
+            var cfgSvc = Services.RequireService<IConfigurationService>();
+            this.arch = cfgSvc.GetArchitecture("x86-protected-32");
+            this.platform = cfgSvc.GetEnvironment("ms-dos")
+                .Load(Services, arch);
+
             var rdr = new LeImageReader(RawImage, FileHeaderOffset);
             var fileHeader = rdr.ReadStruct<FileHeader>();
             var image = new MemoryArea(Address.Ptr32(fileHeader.base_load_offset), new byte[fileHeader.memory_requirements]);
             var w = new LeImageWriter(image.Bytes);
+
             if ((fileHeader.flags & 1) != 0)
             {
-                rdr = new LeImageReader(RawImage, fileHeader.offset_load_image);
-                while (rdr.TryReadUInt16(out ushort us))
+                rdr = new LeImageReader(RawImage, FileHeaderOffset + fileHeader.offset_load_image);
```

Comment:
It would be fantastic to have a HTTP link to the description of this compression format used here, as a comment.

Reply:
Haven't been able to locate anything so far. The open watcom project hasn't implemented packing for Pharlap output yet either.